src/exposurelog/shared_state.py

```diff
@@ -110,16 +109,21 @@ def __init__(self) -> None:
                 f"SITE_ID={site_id!r} too long; max length={SITE_ID_LEN}"
             )
 
-        self.registries: list[lsst.daf.butler.Registry] = []
+        butler_repositories: dict[int, str] = {}
         for i in range(self.num_registries):
-            uri_attr_name = f"butler_uri_{i + 1}"
+            repository_number = i + 1
+            uri_attr_name = f"butler_uri_{repository_number}"
             # The first butler URI is required.
             default = None if i == 0 else ""
             butler_uri = get_env(uri_attr_name.upper(), default=default)
             setattr(self, uri_attr_name, butler_uri)
             if butler_uri != "":
-                butler = lsst.daf.butler.Butler(butler_uri)
-                self.registries.append(butler.registry)
+                butler_repositories[repository_number] = butler_uri
+        self.butler_factory = ButlerFactory(butler_repositories)
+        # Connect to each Butler repository to ensure the configuration is valid.
+        for repository in self.butler_factory.repositories:
+            self.butler_factory.get_butler(repository)
```

Comment:
Can this iteration rise an exception? If that's so, should that be handled perhaps?

Reply:
The intent was to match the behavior of the existing code, which would raise an exception here for failure to start up the Butler connection.  However, I think I might remove this entirely, because it will raise on a transient failure to reach the Butler DB.  

The trade-off is that we don't find out immediately on startup if something is wrong with the Butler, but endpoints that don't use the Butler will work even if Butler is unavailable.